Ginger/GingerCore/ALM/OctaneCore.cs

```diff
@@ -226,5 +365,520 @@ public override bool IsServerConnected()
+                            int stepIndx = tc.Steps.IndexOf(step) + 1;
+                            ActivityIdentifiers actIdent = (ActivityIdentifiers)tcActivsGroup.ActivitiesIdentifiers.Where(x => x.ActivityExternalID == step.StepID).FirstOrDefault();
+                            if (actIdent == null || actIdent.IdentifiedActivity == null) break;//something wrong- shouldn't be null
+                            Activity act = (Activity)actIdent.IdentifiedActivity;
+                            int groupActIndx = tcActivsGroup.ActivitiesIdentifiers.IndexOf(actIdent);
+                            int bfActIndx = busFlow.Activities.IndexOf(act);
+
+                            //set it in the correct place in the group
+                            int numOfSeenSteps = 0;
+                            int groupIndx = -1;
+                            foreach (ActivityIdentifiers ident in tcActivsGroup.ActivitiesIdentifiers)
+                            {
+                                groupIndx++;
+                                if (string.IsNullOrEmpty(ident.ActivityExternalID) ||
+                                        tc.Steps.Where(x => x.StepID == ident.ActivityExternalID).FirstOrDefault() == null)
+                                    continue;//activity which not originally came from the TC
+                                numOfSeenSteps++;
+
+                                if (numOfSeenSteps >= stepIndx) break;
+                            }
+                            ActivityIdentifiers identOnPlace = (ActivityIdentifiers)tcActivsGroup.ActivitiesIdentifiers[groupIndx];
+                            if (identOnPlace.ActivityGuid != act.Guid)
+                            {
+                                //replace places in group
+                                tcActivsGroup.ActivitiesIdentifiers.Move(groupActIndx, groupIndx);
+                                //replace places in business flow
+                                busFlow.Activities.Move(bfActIndx, startGroupActsIndxInBf + groupIndx);
+                            }
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        Reporter.ToLog(eLogLevel.ERROR, $"Method - {MethodBase.GetCurrentMethod().Name}, Error - {ex.Message}", ex);
+                        //failed to re order the activities to match the tc steps order, not worth breaking the import because of this
+                    }
+                }
+
+                //Add the BF variables (linked variables)
+                if (busVariables.Keys.Count > 0)
+                {
+                    foreach (KeyValuePair<string, string> var in busVariables)
+                    {
+                        //add as String param
+                        VariableString busVar = new VariableString();
+                        busVar.Name = var.Key;
+                        ((VariableString)busVar).InitialStringValue = var.Value;
```

Comment:
Codacy found an issue: [Remove this unnecessary cast to 'VariableString'.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6317725)